bika/lims/browser/client/workflow.py

```diff
@@ -135,6 +135,10 @@ def __call__(self):
                     self.context.plone_utils.addPortalMessage(message, 'info')
             if not message:
                 message = _('No changes made.')
+                # Make the message more specific if the reason for not
+                # transitioning is that no Sampler has been selected
+                if not Sampler:
```

Comment:
Note `Sampler` variable might not be initialized (e.g., if the action is not `sample`), so an error like `NameError: name 'Sampler' is not defined` could happen. Also note that even if `Sampler` is initialized, more than one object can be processed here, so you will get the message "Sampler is required for the Sampling transition" if the Sampler was not set for the last iterated item only.

A solution for this would be to add the failure messages to a list while iterating over the objects to be transitioned (loop starting at L61) and join all them later here.

Reply:
@xispa I've changed the code a bit so that a failure message is logged for each `sample` without `Sampler`. Furthermore, I would say thee is no risk in `Sampler` not being initialized because:
1- This lines of code will be only executed if the action is `sample`. See: https://github.com/juangallostra/senaite.core/blob/5c3421e25133ef4d899ee4ac9fa22c358b825220/bika/lims/browser/client/workflow.py#L57 
2- The `Sampler` is checked with the call `sample.getSampler()` which I think returns an empty string if there is no `Sampler`. So I would say it is a safe way of checking for the `Sampler`. As a side note, if the sampler came with the request but wasn't assigned yet to the sample, it will have been assigned already when performing this check. See: https://github.com/juangallostra/senaite.core/blob/5c3421e25133ef4d899ee4ac9fa22c358b825220/bika/lims/browser/client/workflow.py#L76-L83